x-pack/plugins/observability/public/components/app/burn_rate_rule_editor/burn_rate_rule_editor.tsx

```diff
@@ -5,89 +5,136 @@
 
   const onLongWindowDurationChange = (duration: Duration) => {
     setLongWindowDuration(duration);
-    const longWindowdurationInMinutes = toMinutes(duration);
-    const shortWindowDurationValue = Math.floor(longWindowdurationInMinutes / 12);
+    const longWindowDurationInMinutes = toMinutes(duration);
+    const shortWindowDurationValue = Math.floor(longWindowDurationInMinutes / 12);
     setShortWindowDuration({ value: shortWindowDurationValue, unit: 'm' });
   };
 
   const onBurnRateChange = (value: number) => {
     setBurnRate(value);
+    setRuleParams('burnRateThreshold', value);
   };
 
   const onSelectedSlo = (slo: SLO | undefined) => {
     setSelectedSlo(slo);
+    setRuleParams('sloId', slo?.id);
   };
 
   useEffect(() => {
     if (selectedSlo) {
       const sloDurationInMinutes = toMinutes(selectedSlo.timeWindow.duration);
       const longWindowDurationInMinutes = toMinutes(longWindowDuration);
-      setMaxBurnRate(Math.floor(sloDurationInMinutes / longWindowDurationInMinutes));
-    } else {
-      setMaxBurnRate(1);
+      const maxBurnRateThreshold = Math.floor(sloDurationInMinutes / longWindowDurationInMinutes);
+      setMaxBurnRate(maxBurnRateThreshold);
     }
   }, [longWindowDuration, selectedSlo]);
 
+  useEffect(() => {
+    setRuleParams('longWindow', longWindowDuration);
+    setRuleParams('shortWindow', shortWindowDuration);
+  }, [shortWindowDuration, longWindowDuration, setRuleParams]);
+
+  useEffect(() => {
+    setRuleParams('burnRateThreshold', burnRate);
+    setRuleParams('maxBurnRateThreshold', maxBurnRate);
+  }, [burnRate, maxBurnRate, setRuleParams]);
+
   return (
     <EuiFlexGroup direction="column">
       <EuiFlexGroup direction="row">
         <EuiFlexItem>
-          <EuiFormRow label="Select SLO" fullWidth>
-            <SloSelector onSelected={onSelectedSlo} />
-          </EuiFormRow>
+          <SloSelector initialSlo={selectedSlo} onSelected={onSelectedSlo} errors={errors.sloId} />
         </EuiFlexItem>
       </EuiFlexGroup>
 
       <EuiFlexGroup direction="row">
         <EuiFlexItem>
           <LongWindowDuration
             initialDuration={longWindowDuration}
+            shortWindowDuration={shortWindowDuration}
             onChange={onLongWindowDurationChange}
+            errors={errors.longWindow}
           />
         </EuiFlexItem>
         <EuiFlexItem>
-          <BurnRate maxBurnRate={maxBurnRate} onChange={onBurnRateChange} />
+          <BurnRate
+            initialBurnRate={burnRate}
+            maxBurnRate={maxBurnRate}
+            onChange={onBurnRateChange}
+            errors={errors.burnRateThreshold}
+          />
         </EuiFlexItem>
       </EuiFlexGroup>
 
+      <EuiFlexGroup direction="row">
+        {selectedSlo && longWindowDuration?.value > 0 && burnRate >= 1 && (
+          <EuiFlexItem>
+            <EuiText size="s" color="subdued">
+              {getErrorBudgetExhaustionText(
+                numeral(
```

Comment:
Perhaps a good idea to extract this to a named function so its easier to understand what is being calculated